corehq/ex-submodules/pillowtop/processors/elastic.py

```diff
@@ -52,12 +53,14 @@ class ElasticProcessor(PillowProcessor):
       - ES
     """
 
-    def __init__(self, elasticsearch, index_info, doc_prep_fn=None, doc_filter_fn=None):
+    def __init__(self, elasticsearch, index_info, doc_prep_fn=None, doc_filter_fn=None,
+            skip_doc_exists_check=False):
         self.doc_filter_fn = doc_filter_fn or noop_filter
         self.elasticsearch = elasticsearch
         self.es_interface = ElasticsearchInterface(self.elasticsearch)
         self.index_info = index_info
         self.doc_transform_fn = doc_prep_fn or identity
+        self.skip_doc_exists_check = skip_doc_exists_check
```

Comment:
I don't see how we'd be able to set this at the top level? Surely this would be based on individual docs?

Reply:
This setting can be passed down from app_process.yml globally for all docs. I am not adding any heuristics to skip individual docs at this point. Because the pillow processes in bulk, the check for docs shouldn't be a big overhead. If we find that this results in big throughput loss, we can set this flag to True to skip the existence check for all pillow-insertions. See my more detailed comment [here](https://github.com/dimagi/commcare-hq/pull/28574/#issuecomment-703527975)

Reply:
We could also add that heuristic right now.

I am not sure if there is a perfect heuristic that guarantees that a particular doc was never inserted into ES. One heuristic I can think of is `edited_on` is None and `server_modified_on` is within a day or two. This doesn't work if we are reprocessing the pillow for any reason, but in that case, we have `skip_doc_exists_check` set to False explicitly in the pillow config.